src/java/org/apache/cassandra/io/compress/CompressedSequentialWriter.java

```diff
@@ -246,12 +248,12 @@ public synchronized void resetAndTruncate(DataPosition mark)
             compressed = compressor.preferredBufferType().allocate(chunkSize);
         }
 
-        try
+        try(FileChannel readChannel = FileChannel.open(getFile().toPath(), StandardOpenOption.READ))
```

Comment:
If I understand what you're asking (perhaps not), I think it's all right too, at least I  don't know of any reason multiple filechannel's cannot be opened on the same file; I did some searching and could not find anything to think otherwise. 
This is ported from 6.8-cndb, so it's in use today; tests are passing...